src/System.Net.Primitives/src/System/Net/SocketException.Unix.cs

```diff
@@ -2,17 +2,36 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
 using System.ComponentModel;
 
 namespace System.Net.Sockets
 {
     public partial class SocketException : Win32Exception
     {
-        internal SocketException(SocketError errorCode, uint platformError)
-            : this(errorCode)
+        /// <summary>Creates a new instance of the <see cref='System.Net.Sockets.SocketException'/> class with the default error code.</summary>
+        public SocketException() : this(Interop.Sys.GetLastErrorInfo())
```

Comment:
Is this constructor required? I remember deliberately removing any constructor that uses the moral equivalent of `GetLastWin32Error` because it doesn't interact well with the Sockets PAL (e.g. the error we want to return is often not the same as the error returned by `GLW32E`)